Mage.Sets/src/mage/cards/m/MiragePhalanx.java

```diff
@@ -0,0 +1,108 @@
+ */
+public class MiragePhalanx extends CardImpl {
+
+    private static final String ruleText =
+            "As long as {this} is paired with another creature, each of those creatures has " +
+                    "\"At the beginning of combat on your turn, create a token that’s a copy of this creature, " +
+                    "except it has haste and loses soulbond. " +
+                    "Exile it at end of combat.\"";
+
+    public MiragePhalanx(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{R}{R}");
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.SOLDIER);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Soulbond
+        this.addAbility(new SoulbondAbility());
+
+        // As long as Mirage Phalanx is paired with another creature, each of those creatures has
+        // “At the beginning of combat on your turn, create a token that’s a copy of this creature,
+        // except it has haste and loses soulbond.
+        // Exile it at end of combat.”
+        Ability ability = new BeginningOfCombatTriggeredAbility(new MiragePhalanxEffect(), TargetController.YOU, false);
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityPairedEffect(ability, ruleText)));
+    }
+
+    private MiragePhalanx(final MiragePhalanx card) {
+        super(card);
+    }
+
+    @Override
+    public MiragePhalanx copy() {
+        return new MiragePhalanx(this);
+    }
+}
+
+class MiragePhalanxEffect extends OneShotEffect {
+    MiragePhalanxEffect() {
+        super(Outcome.PutCreatureInPlay);
+        this.staticText = "create a token that’s a copy of this creature, " +
+                "except it has haste and loses soulbond. " +
+                "Exile it at end of combat.";
+    }
+
+    private MiragePhalanxEffect(final MiragePhalanxEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) { return false; }
+
+        // Create the token, it has haste
+        CreateTokenCopyTargetEffect tokenCopyEffect = new CreateTokenCopyTargetEffect(source.getControllerId(), null, true);
+        tokenCopyEffect.setTargetPointer(new FixedTarget(permanent, game));
+        tokenCopyEffect.apply(game, source);
+        if (tokenCopyEffect.getAddedPermanent().isEmpty()) { return false; }
+
+        // Change abilities on the token
+        for (Permanent addedToken : tokenCopyEffect.getAddedPermanent()) {
+            // It loses soulbond
+            for (Ability ability : addedToken.getAbilities()) {
+                if (ability instanceof SoulbondAbility) {
+                    LoseAbilityTargetEffect loseSoulbondEffect = new LoseAbilityTargetEffect(ability);
+                    loseSoulbondEffect.setTargetPointer(new FixedTarget(addedToken.getId()));
```

Comment:
you should have the `game` argument for `FixedTarget` even though it's a token

Reply:
Hmm. Another issue with that line. It doesn't seem to work properly. The token copy of Mirage Phalanx no longer has Soulbound listed, but it still triggers when it and the other token enter the battlefield tokens...

Reply:
ok so I looked into it and soulbond is removed from the token before it's created according to [rules 707.9b](https://yawgatog.com/resources/magic-rules/#R7079b) because "has" and "loses" are both after "except"

Reply:
So that sounds like I have to edit `CreateTokenCopyTargetEffect`.

Reply:
unfortunately yes, I think I would add `private final Set<Class<? extends Ability>> clazzesToRemove = new HashSet<>();` and a method for adding ability classes to it and then use that to remove abilities from the token as it's being initialized

Reply:
Looking at `CreateTokenCopyTargetEffect`, it looks like the copy constructor is not copying `this.additionalAbilities`, `this.counter`, `this.numberOfCounters`, and `this.savedPermanent`.

There's no note about it. Is this intentional?

Reply:
No, I'm pretty sure it's a mistake but it doesn't come up in practice because those attributes aren't used in situations where the effect would get copied. Nice catch.

Reply:
Okay, I will fix that with a commit here